packages/rest/src/router/trie-router.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright IBM Corp. 2017, 2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {RouteEntry, createResolvedRoute} from './routing-table';
+import {Trie} from './trie';
+import {Request} from '../types';
+import {inspect} from 'util';
+import {BaseRouter} from './router-base';
+
+const debug = require('debug')('loopback:rest:router:trie');
+
+/**
+ * Router implementation based on trie
+ */
+export class TrieRouter extends BaseRouter {
```

Comment:
AFAICT, there are no test for TrieRouter and the code coverage of this file is ~20% - see https://coveralls.io/builds/19364099/source?filename=packages/rest/src/router/trie-router.ts

Please add some tests to ensure this code works as intended.

Considering how little time we have before 4.0 GA, it may be better to leave `TrieRoute` out of scope of this pull request.